### PR TITLE
Implements learned filter cascade

### PR DESCRIPTION
**Pull request by pgasawa**
- Adds learned filter cascade code from research experiments
- Updates op_examples and github tests for the filter cascade operations

### CODE REVIEW

File: .github/tests/lm_tests.py [hunk 1]
```diff
@@ -21,35 +21,83 @@ def test_filter_operation(setup_models):
...
+        learn_cascade_threshold_sample_percentage=0.5,
+        recall_target=0.9,
+        precision_target=0.9,
+        failure_probability=0.2,
+        return_stats=True,
+    )
+
+    assert "I am really excited to go to class today!" in filtered_df["Text"].values
+    assert "I am very sad" not in filtered_df["Text"].values
+    assert stats["filters_resolved_by_helper_model"] > 0, stats
```
Comment (sidjha1) [inline .github/tests/lm_tests.py:100]: Just a simple question for my own understanding - is `assert stats["filters_resolved_by_helper_model"] > 0` ever going to be flaky depending on the large/small model. Or is this super safe?

File: lotus/sem_ops/cascade_utils.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
+import numpy as np
+
+import lotus
+
+
+def importance_sampling(
+    proxy_scores: list[float],
+    sample_percentage: float,
+):
```
Comment (sidjha1) [inline lotus/sem_ops/cascade_utils.py:9]: Would be helpful to have output types here

**Review by sidjha1 — APPROVED** [review]
Thanks a lot @pgasawa, looks great!

File: .github/tests/lm_tests.py (hunk 1 above)
Comment (pgasawa) [reply .github/tests/lm_tests.py:100]: Good question -- I think achieving >0 helper calls with a .9/.9 targets for this task should be simple enough for language models that I really wouldn't expect it to be flaky. Worst case if it comes to it the line can be removed, though I'd personally be a bit surprised if it failed.

Comment (sidjha1) [reply .github/tests/lm_tests.py:100]: Thanks for clarifying, let's keep it then

File: lotus/sem_ops/cascade_utils.py (hunk 2 above)
Comment (pgasawa) [reply lotus/sem_ops/cascade_utils.py]: Added for all util funcs in the file!

**Pull request merged**